engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/transform/DeploymentTransformer.java

```diff
@@ -150,6 +152,19 @@ private DeploymentResourceTransformer getResourceTransformer(final String resour
         .orElse(UNKNOWN_RESOURCE);
   }
 
+  /**
+   * Exception that can be thrown during processing of a command, in case the engine could not
+   * subscribe to an event. This exception can be handled by the processor in {@link
+   * io.camunda.zeebe.engine.processing.streamprocessor.CommandProcessor#tryHandleError(TypedRecord,
+   * Throwable)}.
+   */
```

Comment:
❌ This documentation is not entirely correct, specifically the part:
>in case the engine could not subscribe to an event

Instead, let's document that this can be thrown whenever the resource cannot be transformed successfully. This allows the platform to rollback any changes the engine made, and that it can be handled in the `DeploymentCreateProcessor#tryHandleError` method.